src/util.jl

```diff
@@ -29,46 +30,58 @@ export  unwrap!,
         shiftin!
 
 """
-    unwrap!(m, dim=ndims(m); range=2pi)
+    unwrap!(y, m; dims=nothing, range=2pi)
 
-In-place version of unwrap(m, dim, range)
+Unwrap `m` storing the result in `y`, see [`unwrap`](@ref).
 """
-function unwrap!(m::Array{T}, dim::Integer=ndims(m); range::Number=2pi) where T<:AbstractFloat
-    thresh = range / 2
-    if size(m, dim) < 2
-        return m
+function unwrap!(y::AbstractArray{<:Any,N}, m::AbstractArray{<:Any,N}; dims=nothing, range::Number=2pi) where {N}
+    if dims === nothing
+        if N != 1
+            throw(ArgumentError("`unwrap!`: required keyword parameter dims missing"))
+        end
+        dims = 1
     end
-    for i = 2:size(m, dim)
-        d = selectdim(m, dim, i:i) - selectdim(m, dim, i-1:i-1)
-        slice_tuple = ntuple(n->(n==dim ? (i:i) : (1:size(m,n))), ndims(m))
-        offset = floor.((d.+thresh) / (range)) * range
-#        println("offset: ", offset)
-#        println("typeof(offset): ", typeof(offset))
-#        println("typeof(m[slice_tuple...]): ", typeof(m[slice_tuple...]))
-#        println("slice_tuple: ", slice_tuple)
-#        println("m[slice_tuple...]: ", m[slice_tuple...])
-        m[slice_tuple...] = m[slice_tuple...] - offset
+    if dims isa Integer
+        Compat.accumulate!(unwrap_kernel(range), y, m, dims=dims)
+    else
+        # this is where #198 can hook in
+        throw(ArgumentError("`unwrap!`: Invalid dims specified: $dims"))
     end
-    return m
+    return y
 end
 
+unwrap_kernel(range=2π) = (x, y) -> y - round((y - x) / range) * range
+
+@deprecate(unwrap!(m::Vector{T}; range::Number=2pi) where T<:AbstractFloat,
+    unwrap!(m, m, range=range))
```

Comment:
remove this deprecation if you end up going with the in-place fallback suggestion above.